src/self_driving/pilot/pilot.cpp

```diff
@@ -38,11 +40,27 @@ void Pilot::PerformPlanning() {
   forecast_ = std::make_unique<WorkloadForecast>(workload_forecast_interval_);
 
   metrics_thread_->PauseMetrics();
-  ExecuteForecast();
+  std::vector<std::pair<const std::string, catalog::db_oid_t>> best_action_seq;
+  Pilot::ActionSearch(&best_action_seq);
   metrics_thread_->ResumeMetrics();
 }
 
-void Pilot::ExecuteForecast() {
+void Pilot::ActionSearch(std::vector<std::pair<const std::string, catalog::db_oid_t>> *best_action_seq) {
+  auto num_segs = forecast_->GetNumberOfSegments();
+  auto end_segment_index = std::min(action_planning_horizon_ - 1, num_segs - 1);
+  std::vector<std::unique_ptr<planner::AbstractPlanNode>> plans;
+  PilotUtil::GetQueryPlans(common::ManagedPointer(this), common::ManagedPointer(forecast_), end_segment_index, &plans);
+  auto mcst = pilot::MonteCarloTreeSearch(common::ManagedPointer(this), common::ManagedPointer(forecast_), plans,
+                                          end_segment_index);
+  mcst.BestAction(simulation_number_, best_action_seq);
```

Comment:
For now, print out information (e.g., which interval, the SQL string) about the selected actions with `LOG_INFO`.